app/controllers/AdminUsers/rewards_controller.rb

```diff
@@ -38,6 +40,37 @@ def destroy
       redirect_to admin_users_rewards_path, notice: 'Reward was successfully deleted.' if reward.destroy
     end
 
+    def new_import_from_csv
+      render :new_import_from_csv, locals: { reward: Reward.new }
+    end
+
+    def import_from_csv
+      if params[:reward][:file].nil?
+        redirect_to new_import_from_csv_admin_users_rewards_path, alert: 'Please select a file'
+      elsif File.extname(params[:reward][:file]) != '.csv'
+        redirect_to new_import_from_csv_admin_users_rewards_path, alert: 'Only .csv files are allowed.'
+      else
+        begin
+          ActiveRecord::Base.transaction do
+            columns = %i[title description price category_id]
+            values = []
+            CSV.foreach(params[:reward][:file].path, headers: true) do |row|
+              category = Category.find_or_create_by(title: row[3].capitalize)
+              values << [row[0], row[1], row[2], category.id]
+            end
+            Reward.import! columns, values, validate: true,
```

Comment:
TBH, I'm not sure if using `activerecord-import` gem helps here - you still have to find category and people not familiar with it would need to read how it specifically works (eg. what `conflict_target` option does). Without `activerecord-import` you could have:
```
reward = Reward.find_or_initialize_by(title: row['Title'])
reward.description = row['Description']
reward.price = row['Price']
reward.category = Category.find_or_create_by!(title: row['Category'])
reward.save!
```
Usually, it's better to use build-in functions and not add third-party dependencies unless really necessary.

Reply:
Understood! I have already removed `activerecord-import` gem and put in this place the code suggested by you. Thanks! 👍